Remotion/ObjectBinding/Web/UI/Controls/BocList.cs

```diff
@@ -1886,7 +1902,9 @@ public BocListSortingOrderEntry[] GetSortingOrder ()
             }
             else
             {
-              var newIndex = Array.IndexOf(columns, entry.Column);
+              var newIndex = entry.Column is BocColumnDefinition bocColumnDefinition
+                ? columns.IndexOf(bocColumnDefinition)
+                : -1;
```

Comment:
I have change all the usages from `ReadOnlyCollection<T>` to `IReadOnlyList<T>` and introduced an extension method for the `IndexOf<T>`. I would keep the existing `is BocColumnDefinition` pattern over the hardcast because `column.Entry` is nullable. Using `is` will prevent an unnecessary `IndexOf` in the case of `null`.

Reply:
> I have change all the usages from ReadOnlyCollection<T> to IReadOnlyList<T> and introduced an extension method for the IndexOf<T>. I would keep the existing is BocColumnDefinition pattern over the hardcast because column.Entry is nullable. Using is will prevent an unnecessary IndexOf in the case of null.

If you want to do "as" instead of "is" we more cloesly match the old semantics. I wouldn't worry about empty sort entries which are already skipped in this implementation. Down in ArrayBuilder, yeah, it could happen, but I don't think special logic is warrented for that case. To me, the soft cast indicates more a "can be another type" not "can also be null" and I don't want to imply the type semantics.